webapp/actions/post_actions.jsx

```diff
@@ -173,6 +175,29 @@ export function createPost(post, files, success) {
         }
     }
 
+    var channelMentionsProp = {};
+    const nameMap = getChannelsNameMapInCurrentTeam(store.getState());
+    var hasMentions = false;
+    while (true) { //eslint-disable-line no-constant-condition
+        const match = CHANNEL_MENTION_PATTERN.exec(post.message);
+        if (!match) {
+            break;
+        }
+        const name = match[3];
+        if (nameMap[name]) {
+            channelMentionsProp[name] = {
+                display_name: nameMap[name].display_name
+            };
```

Comment:
Does this mean that if the channel display name is changed, posts made before the change will still contain the old display name in channel autolinks?

Reply:
Only for people who aren't in that channel. If the client has a more up-to-date name for the channel, it will take precedence.

Reply:
Seems a reasonable tradeoff. +1 from me.